idxmgmt/ilm/policy.go

```diff
@@ -0,0 +1,97 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ilm
+
+var eventPolicies = map[string]map[string]interface{}{
+	"error":       PolicyKeepMedium,
+	"span":        PolicyKeepMedium,
+	"transaction": PolicyKeep,
+	"metric":      PolicyKeep,
+}
+
+var PolicyKeep = map[string]interface{}{
+	"policy": map[string]interface{}{
+		"phases": map[string]interface{}{
+			"hot": map[string]interface{}{
+				"actions": map[string]interface{}{
+					"rollover": map[string]interface{}{
+						"max_size": "50gb",
+						"max_age":  "7d",
+					},
+					"set_priority": map[string]interface{}{
+						"priority": 100,
+					},
+				},
+			},
+			"warm": map[string]interface{}{
```

Comment:
is a warm readonly phase needed if it's going to be frozen immediately anyway?  I might be misunderstanding how this policy works

Reply:
You are right, I misunderstood the `min_age` param to always be relative to the last phase, but as stated in the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/_timing.html) it actually refers to creation or rollover time:
> min_age is usually the time elapsed from the time the index is created. If the index is rolled over, then min_age is the time elapsed from the time the index is rolled over. The intention here is to execute following phases and actions relative to when data was written last to a rolled over index.

I'll update accordingly.